### PR TITLE
max path supported bump

### PR DESCRIPTION
**Pull request by nitesh3108**
# Description
A few sentences describing the overall goals of the pull request's commits.

# GitHub Issues
List the GitHub issues impacted by this PR:

| GitHub Issue # |
| -------------- |
|https://github.com/dell/csm/issues/298|

# Checklist:

- [x] I have performed a self-review of my own code to ensure there are no formatting, vetting, linting, or security issues
- [x] I have verified that new and existing unit tests pass locally with my changes
- [x] I have not allowed coverage numbers to degenerate
- [ ] I have maintained at least 90% code coverage
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] Backward compatibility is not broken

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Please also list any relevant details for your test configuration

Ran functional automation suit on top of k8s-1.24 for CSI-PowerScale, all tests are passing.

### CODE REVIEW

**Review by boyamurthy — COMMENTED** [review]
Please fix formatting issues.

**Review by boyamurthy — APPROVED** [review]
LGTM.

File: middleware/specvalidator/spec_validator.go [hunk 1]
```diff
@@ -772,7 +772,9 @@ func validateVolumeCapabilitiesArg(
 }
 
 const (
-	maxFieldString = 128
+        //Strings such as 'path' 'stagingTargetPath' 'targetpath' 'mountpath' etc.
```
Comment (bpjain2004) [inline middleware/specvalidator/spec_validator.go]: For these strings isn't maxPathLimit available?

File: middleware/specvalidator/spec_validator.go [hunk 2]
```diff
@@ -772,7 +772,9 @@ func validateVolumeCapabilitiesArg(
...
+        //At a minimun, accepted max path length of atleast 192 to keep more safer backword compat.
+	maxFieldString = 192
```
Comment (bpjain2004) [inline middleware/specvalidator/spec_validator.go:777]: why don't we have an env for this limit also similar to EnvVarMaxPathLimit ? This way you can set the proper limit based on k8s version at the driver level. Also, this way you don't need to make these changes in gocsi for every release of the driver/k8s

File: middleware/specvalidator/spec_validator.go [hunk 3]
```diff
@@ -772,7 +772,9 @@ func validateVolumeCapabilitiesArg(
 }
 
 const (
-	maxFieldString = 128
+	//Strings such as 'path' 'stagingTargetPath' 'targetpath' 'mountpath' etc.
+	//At a minimun, accepted max path length of atleast 192 to keep more safer backword compat.
```
Comment (coulof) [inline middleware/specvalidator/spec_validator.go:776]: typos

> At a minimum, accepted max path length of at least 192 to keep safer backward compatibility.

File: middleware/specvalidator/spec_validator.go [hunk 4]
```diff
@@ -772,7 +772,9 @@ func validateVolumeCapabilitiesArg(
...
+	maxFieldString = 192
```
Comment (coulof) [inline middleware/specvalidator/spec_validator.go:777]: As Slack discussed, applying such a limit to all strings is too limited.
Paths can have different maximum lengths depending on the technology (4096 on Linux, 1023 on PowerScale, etc.).

Can you open a ticket so we keep track of this and we redesign that interface ?

Thx

**Review by coulof — DISMISSED** [review]

File: middleware/specvalidator/spec_validator.go (hunk 4 above)
Comment (randeepsharma) [reply middleware/specvalidator/spec_validator.go:777]: Raised a feature to provide option to configure fields. https://github.com/dell/csm/issues/310

File: middleware/specvalidator/spec_validator.go (hunk 1 above)
Comment (randeepsharma) [reply middleware/specvalidator/spec_validator.go]: Raised a feature request to provide option to configure fields. https://github.com/dell/csm/issues/310

Comment (randeepsharma) [reply middleware/specvalidator/spec_validator.go]: maxPathLimit is only for Path of a volume.

**Review by shanmydell — APPROVED** [review]

**Review by randeepsharma — APPROVED** [review]

**Pull request merged**